service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/spring/KeycloakServiceImpl.java

```diff
@@ -228,6 +211,18 @@ private UserRepresentation create(String userName, String password, boolean enab
 
 			user.setCredentials(Arrays.asList(getUserCredentials(password, false)));
 		}
+		if (userTenantInfos == null) {
+			logger.error("create(username={}) no userTenantInfos received", userName);
+			throw new IllegalStateException("Missing userTenantInfos for username " + userName);
+		} else {
+			try {
+				user.singleAttribute(USER_ID_PROPERTY, toTenantInfosString(userTenantInfos));
```

Comment:
what is the end result format this is being persisted as in keycloak? 
it seems to be a string escaped json, which isn't very nice. 
am i missing something or can we do better?

Reply:
it's a JSON, no string escape